### ISSUE TITLE
Asset server panics with "asset paths must have extensions"

### ISSUE THREAD

**Issue by juh9870** [bevyengine/bevy#10703]
## Bevy version

0.12.0

## \[Optional\] Relevant system information
| name | version |
|---|---|
| cargo | 1.73.0 |
| OS | NixOS 23.05.1906.8163a64662b (Stoat) x86_64 |

## What you did

Attempted to load a folder that contains `.directory` files generated by Dolphin file explorer

## What went wrong
Method `get_meta_path` causes asset server to panic with a failed assertion

## Additional information

Debugger screenshot from the moment of panic
![image](https://github.com/bevyengine/bevy/assets/15922601/f1e8d834-2a25-4e11-ac40-356dda4770d1)

Offending code:
https://github.com/bevyengine/bevy/blob/bf30a25efc35e3cd2dd8ad5aee8517eefcf6e72b/crates/bevy_asset/src/io/mod.rs#L248-L258

**Comment by alice-i-cecile** [comment]
Is this fixed with https://github.com/bevyengine/bevy/pull/10153?

**Comment by bushrat011899** [comment]
> Is this fixed with #10153?

Yes I believe this issue will also be resolved. I replaced that `expect` with an `unwrap_or_default` (representing no extension with an empty string)

https://github.com/bushrat011899/bevy/blob/85e39151941e3888df054ba1fd712dfe31a8b092/crates/bevy_asset/src/io/mod.rs#L248-L255
